src/Stratis.Bitcoin.Features.Consensus/CoinViews/Coindb/LeveldbCoindb.cs

```diff
@@ -72,6 +72,41 @@ public void Initialize()
             var options = new Options { CreateIfMissing = true };
             this.leveldb = new DB(options, this.dataFolder);
 
+            // Check if key bytes are in the wrong endian order.
+            HashHeightPair current = this.GetTipHash();
+            byte[] row = this.leveldb.Get(new byte[] { rewindTable }.Concat(BitConverter.GetBytes(current.Height)).ToArray());
+            // Fix the table if required.
+            if (row != null)
+            {
+                // To be sure, check the next height too.
+                byte[] row2 = this.leveldb.Get(new byte[] { rewindTable }.Concat(BitConverter.GetBytes(current.Height - 1)).ToArray());
+                if (row2 != null)
+                {
+                    this.logger.LogInformation("Fixing the coin db.");
+
+                    var rows = new Dictionary<int, byte[]>();
+                    for (int height = 1; height <= current.Height; height++)
+                    {
+                        rows[height] = this.leveldb.Get(new byte[] { rewindTable }.Concat(BitConverter.GetBytes(height)).ToArray());
+                    }
+
+                    using (var batch = new WriteBatch())
+                    {
+                        for (int height = 1; height <= current.Height; height++)
+                        {
+                            batch.Delete(new byte[] { rewindTable }.Concat(BitConverter.GetBytes(height)).ToArray());
```

Comment:
Will this be OK to delete a million records (for example at once)? The same applies to the `put`.

Reply:
In theory there should be no more than `MaxReorg` records when this happens, but one never knows I suppose.